extensions/sendgrid/v1/actions/sendEmailWithTemplate/config/fields.ts

```diff
@@ -0,0 +1,76 @@
+import { isNil, isEmpty } from 'lodash'
+import { z, type ZodTypeAny } from 'zod'
+import {
+  type Field,
+  FieldType,
+  StringType,
+} from '@awell-health/extensions-core'
+
+export const fields = {
+  to: {
+    id: 'to',
+    label: 'To',
+    description: 'The email address of the recipient.',
+    type: FieldType.STRING,
+    stringType: StringType.EMAIL,
+    required: true,
+  },
+  templateId: {
+    id: 'templateId',
+    label: 'Template ID',
+    description: 'The immutable ID of a template that exists in your account.',
+    type: FieldType.STRING,
+    required: true,
+  },
+  subject: {
+    id: 'subject',
+    label: 'Subject',
+    description: 'The subject of your email.',
+    type: FieldType.STRING,
+    required: true,
+  },
+  templateContent: {
+    id: 'templateContent',
+    label: 'Template content',
+    description:
+      'An object (stringified) of template content to send. Dynamic template data is available using Handlebars syntax and this field is a collection of key/value pairs following the pattern "variable_name":"value to insert".',
+    type: FieldType.JSON,
+    required: false,
+  },
+} satisfies Record<string, Field>
+
+interface TemplateContent {
+  [key: string]: string | number | boolean | TemplateContent
```

Comment:
Can't template content be an array as well? See example [here](https://docs.sendgrid.com/ui/sending-email/how-to-send-an-email-with-dynamic-templates#send-a-transactional-email).

Reply:
Didn't know that